Set deprecation warnings as errors (#7235)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/core-upgrade/05-upgrading-to-v1.10.md

- (Introduced, disabled by default) [`validate_macro_args`](/reference/global-configs/behavior-changes#macro-argument-validation). If the flag is set to `True`, dbt will raise a warning if the argument `type` names you've added in your macro YAMLs don't match the argument names in your macro or if the argument types aren't valid according to the [supported types](/reference/resource-properties/arguments#supported-types).

> ### Deprecation warnings as errors

Reviewer: Actually, I think we can remove this as a net-new section in the 1.10 upgrade guide since I've already included it here -> https://github.com/dbt-labs/docs.getdbt.com/pull/7239

> You can now use the [`--warn-error`](/reference/global-configs/warning) flag to set deprecation warnings as errors. Using [`--warn-error-option`](/reference/global-configs/warning#use---warn-error-options-for-targeted-warnings) you can set targeted feature deprecation warnings as errors in combination with other settings to cover scenarios such as:

Author: ```suggestion
You can now use the [`--warn-error`](/reference/global-configs/warning) flag to set deprecation warnings as errors. Using [`--warn-error-option`](/reference/global-configs/warning#use---warn-error-options-for-targeted-warnings), you can set targeted feature deprecation warnings as errors in combination with other settings to cover scenarios such as:
```

> You can now use the [`--warn-error`](/reference/global-configs/warning) flag to set deprecation warnings as errors. Using [`--warn-error-option`](/reference/global-configs/warning#use---warn-error-options-for-targeted-warnings), you can set targeted feature deprecation warnings as errors in combination with other settings to cover scenarios such as:

Author: ```suggestion
You can now use the [`--warn-error`](/reference/global-configs/warnings) flag to set deprecation warnings as errors. Using [`--warn-error-option`](/reference/global-configs/warnings#use---warn-error-options-for-targeted-warnings), you can set targeted feature deprecation warnings as errors in combination with other settings to cover scenarios such as:
```

> You can now use the [`--warn-error`](/reference/global-configs/warnings) flag to set deprecation warnings as errors. Using [`--warn-error-option`](/reference/global-configs/warnings#use---warn-error-options-for-targeted-warnings), you can set targeted feature deprecation warnings as errors in combination with other settings to cover scenarios such as:

Reviewer: You have to use the `--warn-error-options` flag, **not** `--warn-error`

> - Set only deprecation warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`

Reviewer: I _think_ deprecations needs to be capitalized (`Deprecations`) but @QMalcolm can confirm

Reviewer 2: Deprecations _does_ need to be capitalized. This follows the format of different event names which always start with a capital letter.

> - Setting only deprecation warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`

Author: ```suggestion
- Set only deprecation warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`
```

> - Selecting only deprecation warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`

Reviewer 3: ```suggestion
- Setting only deprecation warnings as errors with `--warn-error-options="{'error': ['deprecations']}"`
```

Author: Keeping this as-is to keep it in line with the rest of the page as the -warn flag chooses to change _all_ warnings, the --warn-error-options is used to select which warnings to set as errors

> - Selecting a specific deprecation warning as an error while silencing all others with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`

Reviewer 3: ```suggestion
- Setting a specific deprecation warning as an error while silencing all other deprecations with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`
```

> - Selecting a specific deprecation warning as an error while silencing all others with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`

Author: ```suggestion
- Selecting a specific deprecation warning as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`
```

> - Set a specific deprecation warning as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`

Reviewer: i think we should give an example of a real deprecation name _not_ "OneSpecificDeprecation"

Reviewer 2: How about `PackageRedirectDeprecation`?

> - Set a specific deprecation warning to set as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`

Reviewer 3: ```suggestion
- Set a specific deprecation warning as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`
```

> - Selecting a specific deprecation warning as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`

Author: ```suggestion
- Select a specific deprecation warning to set as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`
```

> - Select a specific deprecation warning to set as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`

Author: ```suggestion
- Set a specific deprecation warning to set as an error while silencing all other deprecation warnings with `--warn-error-options="{'error': ['OneSpecificDeprecation'], 'silence': ['deprecations']}"`
```
